examples/plot_psifci_alg.py

```diff
@@ -0,0 +1,151 @@
+"""
+.. _ex-psifci-algorithm:
+
+=========================================================
+Causal discovery with interventional data - Sachs dataset
+=========================================================
+
+We will analyze the Sachs dataset :footcite:`sachsdataset2005` and reproduce analyses
+from the Supplemental Figure 8 in :footcite:`Jaber2020causal` demonstrating the
+usage of the :class:`dodiscover.constraint.PsiFCI` algorithm for learning causal graphs
+from observational and interventional data.
+
+.. currentmodule:: dodiscover
+"""
+
+
+# %%
+# Authors: Adam Li <adam2392@gmail.com>
+#
+# License: BSD (3-clause)
+
+from pywhy_graphs.viz import draw
+from dodiscover.ci import GSquareCITest
+from dodiscover import PsiFCI, Context, make_context, InterventionalContextBuilder
+
+import pandas as pd
+import bnlearn
+
+import pooch
+
+# %%
+# Pull in the Sachs Dataset
+# -------------------------
+# The Sachs dataset is a famous dataset in causal discovery because of its real-life
+# applicability and access to experimental data that analyzed the causal network of
+# 11 proteins using knockouts and spikings :footcite:`sachsdataset2005`. The pathways
+# for those proteins are already known, so it is an ideal dataset for benchmarking
+# causal discovery algorithms.
+#
+# We will download a preprocessed version of the dataset from the following
+# url: https://www.bnlearn.com/book-crc/code/sachs.interventional.txt.gz
+#
+# Ref: https://erdogant.github.io/bnlearn/pages/html/bnlearn.bnlearn.html#bnlearn.bnlearn.import_example  # noqa
+
+# use pooch to download robustly from a url
+url = "https://www.bnlearn.com/book-crc/code/sachs.interventional.txt.gz"
+file_path = pooch.retrieve(
+    url=url,
+    known_hash="md5:39ee257f7eeb94cb60e6177cf80c9544",
+)
+
+df = pd.read_csv(file_path, delimiter=" ")
+
+# the ground-truth dag is shown here: XXX: comment in when errors are fixed
+ground_truth_dag = bnlearn.import_DAG("sachs", verbose=False)
+fig = bnlearn.plot(ground_truth_dag)
+
+# .. note::
+#    The Sachs dataset has previously been preprocessed, and the steps are described
+#    in bnlearn, at the web-page https://www.bnlearn.com/research/sachs05/.
+print(df.head())
+print(df.shape)
+
+# %%
+# Preprocess the dataset
+# ----------------------
+# Since the data is one dataframe, we need to process it into a form
+# that is acceptable by dodiscover's :class:`constraint.PsiFCI` algorithm. We
+# will form a list of separate dataframes.
+unique_ints = df["INT"].unique()
+
+# get the list of intervention targets and list of dataframe associated with each intervention
+intervention_targets = [df.columns[idx] for idx in unique_ints]
```

Comment:
This example probably requires more extensive explanation. I cannot really follow where this intervention suddenly comes from. Is this already part of the data set? Are there multiple datasets from different interventional environments? Maybe introducing this a bit more in detail can be helpful.

Reply:
I added some additional description about the preprocessed dataset in the start of the example file: 8ef2752

Reply:
@bloebp There is an INT column in the Sachs dataset.  It specifies which of the other columns was a target if intervention in that row